bootstrap.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+"""Bootstrap script for emscripten developers / git users.
+
+After checking out emscripten there are certain steps that need to be
+taken before it can be used.  This script enumerates and automates
+these steps and is able to run just the steps that are needed based
+on the timestamps of various input files (kind of like a dumb version
+of a Makefile).
+"""
+import argparse
+import os
+import shutil
+import sys
+
+__rootdir__ = os.path.dirname(os.path.abspath(__file__))
+sys.path.insert(0, __rootdir__)
+
+STAMP_DIR = os.path.join(__rootdir__, 'out')
+
+from tools import shared, utils
+
+actions = [
+  ('npm packages', 'package.json', [shutil.which('npm'), 'ci']),
+  # TODO(sbc): Remove the checked in entry point files and have them
+  # built on demand by this step.
+  ('create entry points', 'tools/maint/create_entry_points.py', [sys.executable, 'tools/maint/create_entry_points.py']),
```

Comment:
I would prefer for the entry points to remain part of the repository, since if these are generated on demand, it will prevent being able to run `git clean -xdf` to get a checkout back to a clean slate. (one will then have to keep running bootstrap after cleaning before being able to run commands)